app_test.go

```diff
@@ -5,98 +5,212 @@
+	return mux, nil
+}
+
+// header check
+func checkAuth(response http.ResponseWriter, request *http.Request) {
+	authPassword := request.Header.Get(AUTH_HEADER_PASSWORD)
+	authToken := request.Header.Get(AUTH_HEADER_TOKEN)
+	userAndPass := base64.StdEncoding.EncodeToString(
+		[]byte(KINTONE_USERNAME + ":" + KINTONE_USERNAME))
+	if authToken != "" && authToken != KINTONE_API_TOKEN {
+		http.Error(response, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+	} else if authPassword != userAndPass {
+		http.Error(response, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+	}
+
+}
+func checkContentType(response http.ResponseWriter, request *http.Request) {
+	contentType := request.Header.Get("Content-Type")
+	if contentType != "application/json" {
+		http.Error(response, http.StatusText(http.StatusNoContent), http.StatusNoContent)
 	}
 }
 
-func newAppInGuestSpace(appId uint64, guestSpaceId uint64) *App {
-	return &App{
-		Domain:       KINTONE_DOMAIN,
-		User:         KINTONE_USER,
-		Password:     KINTONE_PASSWORD,
-		AppId:        appId,
-		GuestSpaceId: guestSpaceId,
+// handler mux
+func handleResponseForm(response http.ResponseWriter, request *http.Request) {
+	checkAuth(response, request)
+	if request.Method == "GET" {
+		checkContentType(response, request)
+		testData := GetDataTestForm()
+		fmt.Fprint(response, testData.output)
 	}
 }
 
-func TestGetRecord(t *testing.T) {
-	a := newApp(4799)
-	if len(a.Password) == 0 {
-		t.Skip()
+func handleResponseRecordsCursor(response http.ResponseWriter, request *http.Request) {
+	checkAuth(response, request)
+	if request.Method == "GET" {
+		testData := GetDataTestGetRecordsByCursor()
+		fmt.Fprint(response, testData.output)
+	} else if request.Method == "DELETE" {
+		checkContentType(response, request)
+		testData := GetTestDataDeleteCursor()
+		fmt.Fprint(response, testData.output)
+	} else if request.Method == "POST" {
+		checkContentType(response, request)
+		testData := GetTestDataCreateCursor()
+		fmt.Fprint(response, testData.output)
 	}
+}
 
-	if rec, err := a.GetRecord(116); err != nil {
-		t.Error(err)
-	} else {
-		if rec.Id() != 116 {
-			t.Errorf("Unexpected Id: %d", rec.Id())
-		}
-		for _, f := range rec.Fields {
-			if files, ok := f.(FileField); ok {
-				if len(files) == 0 {
-					continue
-				}
-				fd, err := a.Download(files[0].FileKey)
-				if err != nil {
-					t.Error(err)
-				} else {
-					data, _ := ioutil.ReadAll(fd.Reader)
-					t.Logf("%s %d bytes", fd.ContentType, len(data))
-				}
-			}
-		}
+func handleResponseRecordComments(response http.ResponseWriter, request *http.Request) {
+	checkAuth(response, request)
+	if request.Method == "POST" {
+		checkContentType(response, request)
+		testData := GetTestDataAddRecordComment()
+		fmt.Fprint(response, testData.output)
+	} else if request.Method == "DELETE" {
+		checkContentType(response, request)
+		testData := GetDataTestDeleteRecordComment()
+		fmt.Fprint(response, testData.output)
```

Comment:
If they are all used "checkContentType" please move it to out side for common using.